README.md

```diff
@@ -1 +1,1148 @@
+[Info]    Everything is up to date (use --build to force a rebuild)
+[Info]    Creating release...
+[Info]    Pushing images to registry...
+[Info]    Saving release...
+[Success] Deploy succeeded!
+[Success] Release: b9e5ec4b309f91281ecb592028dcea0c
+
+			    \
+			     \
+			      \\
+			       \\
+			        >\/7
+			    _.-(6'  \
+			   (=___._/` \
+			        )  \ |
+			       /   / |
+			      /    > /
+			     j    < _\
+			 _.-' :      ``.
+			 \ r=._\        `.
+			<`\\_  \         .`-.
+			 \ r-7  `-. ._  ' .  `\
+			  \`,      `-.`7  7)   )
+			   \/         \|  \'  / `-._
+			              ||    .'
+			               \\  (
+			                >\  >
+			            ,.-' >.'
+			           <.'_.''
+			             <'
+```
+This will create a new release (visible via the dashboard), and push the image
+directly to the balena Docker registry. Your balenaFin should then download the
+new release and run it. This is useful if you already have an image pre-built
+and just need to upload it.
+
+However, `balena deploy` also allows you to complete the build step as well
+implicitly by not specifying an image to upload. Run the following command in
+the `balena-cli-masterclass` repository:
+```
+$ balena deploy cliApp --build --emulated
+[Info]    Creating default composition with source: /Work/Support/MasterClasses/repos/balena-cli-masterclass
+[Info]    Building for armv7hf/fincm3
+[Build]   Built 1 service in 0:58
+[Build]   main Image size: 213.80 MB
+[Info]    Creating release...
+[Info]    Pushing images to registry...
+[Info]    Saving release...
+[Success] Deploy succeeded!
+[Success] Release: 13f39923c2ddf95fa35a129d8efb5b53
+
+			    \
+			     \
+			      \\
+			       \\
+			        >\/7
+			    _.-(6'  \
+			   (=___._/` \
+			        )  \ |
+			       /   / |
+			      /    > /
+			     j    < _\
+			 _.-' :      ``.
+			 \ r=._\        `.
+			<`\\_  \         .`-.
+			 \ r-7  `-. ._  ' .  `\
+			  \`,      `-.`7  7)   )
+			   \/         \|  \'  / `-._
+			              ||    .'
+			               \\  (
+			                >\  >
+			            ,.-' >.'
+			           <.'_.''
+			             <'
+```
+This forces the `deploy` command to first build (or rebuild if the image already
+exists) the project before pushing it to the Docker registries.
+
+## 6. Using Local Mode to Develop Applications
+
+So far, you've seen how to push code to the balena builders, or to build and
+push images on a development machine. Whilst practical solutions for pre-tested
+code, or for a CI pipeline, this is not a fast workflow for active development
+of an application by an engineer as it involves rebuilding an image and then
+delivering it to the target device.
+
+To make active development of applications easier for an engineer, balena
+devices provisioned with a development image include a device mechanism called
+'Local Mode'. This can be activated easily from the dashboard. Go to your
```

Comment:
Too bad local mode cannot be flipped from the CLI!

Reply:
This is a really good point, I think it should go into `balena-cli`.

Reply:
There's an outstanding issue: https://github.com/balena-io/balena-cli/issues/1304